submission/submission-webapp/src/main/kotlin/ac/uk/ebi/biostd/submission/domain/helpers/SourceGenerator.kt

```diff
@@ -17,57 +21,79 @@ class SourceGenerator(
     private val props: ApplicationProperties,
     private val fireSourceFactory: FireFilesSourceFactory,
 ) {
-    fun userSources(
-        user: SecurityUser,
-        rootPath: String? = null
-    ): FilesSource = ComposedFileSource(userSourcesList(user, rootPath.orEmpty()))
-
     fun submissionSources(requestSources: RequestSources): FilesSource {
-        val (owner, submitter, files, rootPath, submission) = requestSources
-        return ComposedFileSource(submissionSources(owner, submitter, files, rootPath, submission))
-    }
+        val sources = buildList {
```

Comment:
Here I see an issue with this code but I am aware is really open to discussion. Let me share my thoughts. Our final goal as developer is to make code easy to mantain and we do that but creating code easy to undersdant. All others practices i.e code re use , small functions etc IMO they are just side effects of this.

Here you have a function which is split in many small ones (which in theory is good because small functions are good ...) but in my opinion code is harder to read because one need to fo to many places to answer this simple questions "Give x configuration what sources do we use".  Which is not a random question that I just imagine my self but the question that any developer comming to this class will have.

 So in my opinion here we need to break the gold rule (of small function) and favor simplicity by having code all togther. 
 

 ```
     fun submissionSources(requestSources: RequestSources): FilesSource {
        val (owner, submitter, files, rootPath, submission, preferredSources) = requestSources
        val sources = preferredSources.ifEmpty { listOf(USER_SPACE, SUBMISSION, FIRE) }
        return ComposedFileSource(submissionSources(owner, submitter, files, rootPath, submission, sources))
    }

    private fun submissionSources(
        owner: SecurityUser?,
        submitter: SecurityUser?,
        files: List<File>,
        rootPath: String?,
        sub: ExtSubmission?,
        sources: List<PreferredSource>,
    ): List<FilesSource> = buildList {
        add(FilesListSource(files))
        sources.forEach { it ->
            when (it) {
                FIRE -> if (props.persistence.enableFire) add(fireSourceFactory.createFireSource())
                USER_SPACE -> {
                    if (submitter != null) {
                        add(createPathSource(submitter, rootPath))
                        addAll(groupSources(submitter.groupsFolders))
                    }
                    if (owner != null) {
                        add(createPathSource(owner, rootPath))
                        addAll(groupSources(owner.groupsFolders))
                    }
                }
                SUBMISSION -> {
                    if (sub != null) {
                        add(fireSourceFactory.createSubmissionFireSource(sub.accNo, Paths.get("${sub.relPath}/Files")))
                        add(PathFilesSource(Paths.get(props.submissionPath).resolve(sub.relPath).resolve(FILES_PATH)))
                    }
                }
            }
        }
    }
 ```